bundles/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComWindMessage.java

```diff
@@ -1,253 +1,259 @@
-    public List<RFXComValueSelector> getSupportedValueSelectors() throws RFXComException {
-        return supportedValueSelectors;
-    }
-
-}
+/**
+ * Copyright (c) 2010-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.rfxcom.internal.messages;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.openhab.binding.rfxcom.RFXComValueSelector;
+import org.openhab.binding.rfxcom.internal.RFXComException;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.Type;
+import org.openhab.core.types.UnDefType;
+
+/**
+ * RFXCOM data class for temperature and humidity message.
+ *
+ * @author Marc SAUVEUR
+ * @since 1.4.0
+ */
+public class RFXComWindMessage extends RFXComBaseMessage {
+
+    public enum SubType {
+        WTGR800(1),
+        WGR800(2),
+        STR918_WGR918_WGR928(3),
+        TFA(4),
+        UPM_WDS500(5),
+        WS2300(6),
+        WS4500_ET_AL(7),
+
+        UNKNOWN(255);
+
+        private final int subType;
+
+        SubType(int subType) {
+            this.subType = subType;
+        }
+
+        SubType(byte subType) {
+            this.subType = subType;
+        }
+
+        public byte toByte() {
+            return (byte) subType;
+        }
+
+        public static SubType fromByte(int input) {
+            for (SubType c : SubType.values()) {
+                if (c.subType == input) {
+                    return c;
+                }
+            }
+
+            return SubType.UNKNOWN;
+        }
+    }
+
+    private final static List<RFXComValueSelector> supportedValueSelectors = Arrays.asList(RFXComValueSelector.RAW_DATA,
+            RFXComValueSelector.SIGNAL_LEVEL, RFXComValueSelector.BATTERY_LEVEL, RFXComValueSelector.WIND_DIRECTION,
+            RFXComValueSelector.WIND_AVSPEED, RFXComValueSelector.WIND_SPEED, RFXComValueSelector.TEMPERATURE,
+            RFXComValueSelector.CHILL_FACTOR);
+
+    public SubType subType = SubType.UNKNOWN;
+    public int sensorId = 0;
+    public double windDirection = 0;
+    public double windSpeed = 0;
+    public double windAvSpeed = 0;
+    public double temperature = 0; // TFA type only
+    public double chillFactor = 0; // TFA type only
+    public byte signalLevel = 0;
+    public byte batteryLevel = 0;
+
+    public RFXComWindMessage() {
+        packetType = PacketType.WIND;
+    }
+
+    public RFXComWindMessage(byte[] data) {
+        encodeMessage(data);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Consider refactoring this to use a StringBuffer for building up the return value.
Absolutely not required for this PR.